### PR TITLE
PCHR-2379: Optimize notification service and fix pre-commit issues

### PR DESCRIPTION
**Pull request by bmnepali**
## Overview
As per @igorpavlov comment on the optimization of the notification service, this PR adds the optimization along with   some  pre-commit warnings.

## Technical Details
1. Optimized notification service for even ore compact version as follows:
```js
/* eslint-env amd */

(function (CRM) {
  define([
    'common/lodash',
    'common/modules/services'
  ], function (_, services) {
    'use strict';

    services.factory('notificationService', function () {
      return _(['alert', 'success', 'info', 'error'])
        .map(function (type) {
          return [type, function (title, body, options) {
            return CRM.alert(body, title, type, options);
          }];
        })
        .zipObject()
        .value();
    });
  });
}(CRM));
```
2. Adopting support for new angular js style guide and naming conversion for test files:
- Made changes to recognize `*.spec.js` as test spec file  replacing `*_test.js` files in `gulpfile.js` and `karma.conf.js` files.
- Renamed `*_test.js` spec files to `.spec.js`
- Fix pre-commit issues for all the spec files

- [x] Tests Pass

### CODE REVIEW

File: org.civicrm.reqangular/src/common/services/notification.js [hunk 1]
```diff
@@ -1,56 +1,23 @@
+/* globals CRM */
```
Comment (AkA84) [inline org.civicrm.reqangular/src/common/services/notification.js]: you don't need this

File: org.civicrm.reqangular/test/services/notification_test.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* globals CRM, inject */
```
Comment (AkA84) [inline org.civicrm.reqangular/test/services/notification_test.js]: you don't need this either

Comment (AkA84) [inline org.civicrm.reqangular/test/services/notification_test.js]: the file should be notification.spec.js

File: org.civicrm.reqangular/src/common/services/notification.js (hunk 1 above)
Comment (AkA84) [inline org.civicrm.reqangular/src/common/services/notification.js]: also, as per angular style guide, the file should have a `.service.js` suffix

Comment (AkA84) [inline org.civicrm.reqangular/src/common/services/notification.js]: still as per the angular style guide, if the name of the service is a noun, you should append "Service" to it. So it should be "notificationService" i think

File: org.civicrm.reqangular/src/common/services/notification.js [hunk 3]
```diff
@@ -1,56 +1,23 @@
...
-         * Uses CRM.alert to display info notification
-         *
-         * @param {string} title
-         * @param {string} body
-         * @param {object} options
-         */
-        info: function (title, body, options) {
-          return CRM.alert(body, title, 'info', options);
-        },
+      notificationTypes.forEach(function (notificationType) {
```
Comment (AkA84) [inline org.civicrm.reqangular/src/common/services/notification.js]: A more compact variant

```js
services.factory('notification', function () {
  return _(['alert', 'success', 'info', 'error'])
    .map(function (type) {
      return [type, function (title, body, options) {
        return CRM.alert(body, title, type, options);
      }];
    })
    .fromPairs()
    .value();
});
```

File: org.civicrm.reqangular/test/services/notification_test.js (hunk 2 above)
Comment (bmnepali) [inline org.civicrm.reqangular/test/services/notification_test.js]: done!

**Comment by igorpavlov** [comment]
@bmnepali did you see my comment here https://github.com/civicrm/civihr/pull/2014#issuecomment-318421154 ? Does this PR solve this issue?

File: org.civicrm.reqangular/test/services/notification_test.js (hunk 2 above)
Comment (bmnepali) [inline org.civicrm.reqangular/test/services/notification_test.js]: removed!

File: org.civicrm.reqangular/src/common/services/notification.js (hunk 1 above)
Comment (bmnepali) [inline org.civicrm.reqangular/src/common/services/notification.js]: done!

Comment (bmnepali) [inline org.civicrm.reqangular/src/common/services/notification.js]: done!

Comment (bmnepali) [inline org.civicrm.reqangular/src/common/services/notification.js]: done!

**Comment by bmnepali** [comment]
@igorpavlov The support for notification service is already added in PR #2014 and it's already merged and  its working in civihr staging site. This PR is only for the optimization, remaining PR review feedbacks to #2014  and adopting new angular js style guide for tests files. 

cc @AkA84

**Comment by bmnepali** [comment]
Last two commits are after rebase so don't need  a review.

**Comment by bmnepali** [comment]
Rebased to fix merge conflicts.

**Pull request merged**